src/index.js

```diff
@@ -38,158 +38,144 @@ function getClosestPackage(modulePath) {
       if (!closestPackage) {
         return;
       }
 
-      pkg = closestPackage.package;
-      packagePath = closestPackage.path;
+      const pkg = closestPackage.package;
+      const packagePath = closestPackage.path;
 
-      let modulePath = cleanPathRelativeToContext(packagePath);
+      const modulePath = cleanPathRelativeToContext(packagePath);
 
-      let version = pkg.version;
+      const { version } = pkg;
 
       modules[pkg.name] = modules[pkg.name] || [];
 
-      let isSeen = _.find(modules[pkg.name], module => {
-        return module.version === version;
-      });
+      const isSeen = modules[pkg.name].find(pkgModule => pkgModule.version === version);
 
       if (!isSeen) {
-        let entry = { version, path: modulePath };
+        const entry = { version, path: modulePath };
 
-        let issuer =
-          module.issuer && module.issuer.resource
-            ? cleanPathRelativeToContext(module.issuer.resource)
-            : null;
+        const issuer = module.issuer && module.issuer.resource
+          ? cleanPathRelativeToContext(module.issuer.resource)
+          : null;
         entry.issuer = issuer;
 
         modules[pkg.name].push(entry);
       }
     });
 
-    let duplicates = {};
+    const duplicates = {};
 
-    for (let name in modules) {
+    Object.keys(modules).forEach((name) => {
       const instances = modules[name];
 
       if (instances.length <= 1) {
-        continue;
+        return;
       }
 
       let filtered = instances;
       if (!strict) {
         filtered = [];
-        const groups = _.groupBy(instances, instance =>
-          semver.major(instance.version)
-        );
+        const groups = groupBy(instances, instance => semver.major(instance.version)) || {};
 
-        _.each(groups, group => {
-          if (group.length > 1) {
+        Object.keys(groups).forEach((groupKey) => {
+          const group = groups[groupKey];
+          if (group && group.length > 1) {
             filtered = filtered.concat(group);
           }
         });
 
         if (filtered.length <= 1) {
-          continue;
+          return;
         }
       }
 
       if (exclude) {
-        filtered = filtered.filter(instance => {
-          instance = Object.assign({ name }, instance);
-          return !exclude(instance);
-        });
+        filtered = filtered.filter(instance => !exclude(Object.assign({ name }, instance)));
 
         if (filtered.length <= 1) {
-          continue;
+          return;
         }
       }
 
       duplicates[name] = filtered;
-    }
+    });
 
     const duplicateCount = Object.keys(duplicates).length;
 
     if (duplicateCount) {
-      let array = emitError ? compilation.errors : compilation.warnings;
+      const sortedDuplicateKeys = Object.keys(duplicates).sort();
 
-      let i = 0;
+      sortedDuplicateKeys.forEach((name, index) => {
+        const array = emitError || alwaysEmitErrorsFor.indexOf(name) >= 0
```

Comment:
We use the `alwaysEmitErrorsFor` array here to determine whether to throw errors or not.